github/azure-self-hosted-runners/tf/providers.tf

```diff
@@ -0,0 +1,18 @@
+terraform {
+  required_version = ">=0.12"
+
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = "~>3.0"
+    }
+    random = {
+      source  = "hashicorp/random"
+      version = "~>3.0"
+    }
+  }
```

Comment:
Can we use a remote store instead of local file store? https://github.com/confidential-containers/cloud-api-adaptor/blob/ade303e0acddcae9aa94bc753866578215587ae7/ci-infra/azure/providers.tf#L11-L17